inspirehep/modules/records/receivers.py

```diff
@@ -334,3 +335,15 @@ def add_recids_and_validate(sender, json, *args, **kwargs):
     """Ensure that recids are generated before being validated."""
     populate_recid_from_ref(sender, json, *args, **kwargs)
     references_validator(sender, json, *args, **kwargs)
+
+
+@before_record_update.connect
+def check_if_record_is_going_to_be_deleted(sender, *args, **kwargs):
```

Comment:
? Are you sure? `before_record_update()` is called in `invenio_records.api` with:
```python
before_record_update.send(self)
```

so there is no json argument.